pkg/sql/internal.go

```diff
@@ -278,6 +278,11 @@ func (ie *InternalExecutor) initConnEx(
 	var ex *connExecutor
 	var err error
 	if txn == nil {
+		postSetupFn := func(ex *connExecutor) {
+			// Inject any synthetic descriptors into the right after the
+			// internal executor is created.
+			ex.extraTxnState.descCollection.SetSyntheticDescriptors(ie.syntheticDescriptors)
+		}
```

Comment:
It seems we are breaking some assumptions previously made here. 

From what I read, it looks like we should only set `postSetupFn` if `txn != nil` (see [this comment](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/conn_executor.go#L1016-L1019)). I thought to myself why it says this and maybe it's because we should only set extraTxnState of the newly initiated connExecutor from parent internal executor, if parent internal executor is being run in a non-nil txn.

What you've changed here effectively changes that assumption, because you are saying "even when the internal executor isn't running in any existing txn, we should inherit/share the syntheticDescriptors". Why is this?

Finally, why are we only doing this when `txn == nil`? What about the other branch when `txn != nil`? Don't we need to also set the synthetic descriptors there from the parent internal executor?

Reply:
Good point on the non-nil contract. Let's figure if the contract can be break. From what I'm seeing, the contract can be break and there is nothing preventing us from doing that.

For your question on why not doing this in the `txn != nil` branch, it's because if it's running under a txn, you can just inherit the synthetic from the executor's collection. Note that when `txn == nil`, we just create a new collection.

Reply:
Yeah, the non-nil case here is incorrect we shouldn't set synthetic descriptors if they are nil. Additionally, we should set the reset flag here and clean them tup if we decide to set them in the first place.